File: project/src/app/models/investment.ts
import { InvestmentType } from "./investment-type";

export class Investment {
  public id: number;
  public name: string;
  public ticker?: string;
  public type: string;
  public date_Bought: Date;
  public date_Sold?: Date = undefined;
  public status: string;
  public cost: number;
  public revenue?: number = undefined;
  public dividends: number = 0;
  public net_Profit_Percentage?: number;
  public net_Profit?: number;

  constructor(id: number, name: string, type: InvestmentType, date_Bought: Date, status: string, cost: number) {
    this.id = id;
    this.name = name;
    this.type = type.type;
    this.date_Bought = date_Bought;
    this.status = status;
    this.cost = cost;
  }
}

File: API/InvestmentPortfolio/Investment.cs
namespace InvestmentPortfolio
{
    public class Investment
    {
        public int ID { get; set; }
        public string Name { get; set; } = String.Empty;
        public string? Ticker { get; set; }
        public string Type { get; set; } = String.Empty;
        public DateTime Date_Bought { get; set; } = new DateTime();
        public DateTime? Date_Sold { get; set; } = null;
        public string Status { get; set; } = String.Empty;
        public double Cost { get; set; }
        public double? Revenue { get; set; } = null;
        public double? Dividends { get; set; } = 0;
        public double? Net_Profit_Percentage { get; set; }
        public double? Net_Profit { get; set; }
    }
}
